src/nebari/hookspecs.py

```diff
@@ -1,54 +1,37 @@
-import contextlib
-import pathlib
-from typing import Any, Dict, List
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import ClassVar, Optional
 
-import pydantic
-import typer
 from pluggy import HookimplMarker, HookspecMarker
+from pydantic import BaseModel
+from typer import Typer
 
 from nebari import schema
 
 hookspec = HookspecMarker("nebari")
 hookimpl = HookimplMarker("nebari")
 
 
-class NebariStage:
-    name: str = None
-    priority: int = None
+class NebariStage(ABC):
+    name: ClassVar[str]
+    priority: ClassVar[int]
+    input_schema: ClassVar[Optional[BaseModel]] = None
+    output_schema: ClassVar[BaseModel]
 
-    input_schema: pydantic.BaseModel = None
-    output_schema: pydantic.BaseModel = None
-
-    def __init__(self, output_directory: pathlib.Path, config: schema.Main):
+    def __init__(self, output_directory: Path, config: schema.Main):
         self.output_directory = output_directory
         self.config = config
 
-    def render(self) -> Dict[str, str]:
-        return {}
-
-    @contextlib.contextmanager
-    def deploy(
-        self, stage_outputs: Dict[str, Dict[str, Any]], disable_prompt: bool = False
-    ):
-        yield
-
-    def check(
-        self, stage_outputs: Dict[str, Dict[str, Any]], disable_prompt: bool = False
-    ) -> bool:
-        pass
-
-    @contextlib.contextmanager
-    def destroy(
-        self, stage_outputs: Dict[str, Dict[str, Any]], status: Dict[str, bool]
-    ):
-        yield
```

Comment:
Those don't apply for BootstrapStage.

Reply:
Removing these hookspec methods effectively removes these as pluggable methods for downstream extensions.

These methods are not used by the BootstrapStage but are used by NebariTerraformStage. 

Perhaps there is more work needed before we can tear these methods out. Either adding them as hookspecs in another class or something else.